modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_builder.py

```diff
@@ -754,8 +754,10 @@ def _push(self, node):
             and all(isinstance(expr, CalciteInputRefExpr) for expr in node.exprs)
         ):
             # Replace the last CalciteProjectionNode with this one and
-            # translate the input refs.
-            exprs = self.res.pop().exprs
+            # translate the input refs. The `id` attribute is preserved.
+            last = self.res.pop()
+            exprs = last.exprs
+            type(last)._next_id[0] = int(last.id)
```

Comment:
Let's introduce classmethod for `CalciteBaseNode` to hide this implementation detail.